contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -1,26 +1,137 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import { StableMath } from "../utils/StableMath.sol";
 import { VaultCore } from "./VaultCore.sol";
 
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+
 /**
  * @title OETH VaultCore Contract
  * @author Origin Protocol Inc
  */
 contract OETHVaultCore is VaultCore {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
     address public immutable weth;
+    uint256 public wethAssetIndex;
 
     constructor(address _weth) {
         weth = _weth;
     }
 
+    /**
+     * @dev Caches WETH's index in `allAssets` variable.
+     *      Reduces gas usage by redeem by caching that.
+     */
+    function cacheWETHAssetIndex() external onlyGovernor {
+        uint256 assetCount = allAssets.length;
+        for (uint256 i = 0; i < assetCount; ++i) {
+            if (allAssets[i] == weth) {
+                wethAssetIndex = i;
+                break;
+            }
+        }
+
+        require(allAssets[wethAssetIndex] == weth, "Invalid WETH Asset Index");
+    }
+
     // @inheritdoc VaultCore
     function mint(
         address _asset,
         uint256 _amount,
-        uint256 _minimumOusdAmount
+        uint256
```

Comment:
Let's actually check these minimum amounts on both mints and redeems. This keeps integrations from using bogus high values, which would prevent us from using this for real at some point in the future.